ir/src/transition_stmts/graph.rs

```diff
@@ -87,8 +88,9 @@ impl AlgebraicGraph {
     /// added recursively to reuse existing matching nodes.
     pub(super) fn insert_expr(
         &mut self,
-        symbol_table: &SymbolTable,
+        symbol_table: &mut SymbolTable,
```

Comment:
I don't think the symbol table needs to be mutable, does it? It doesn't seem to be changed anywhere inside this module

Reply:
Thanks, Fixed. Forgot to change it after refactor.